model/qti/identifierGenerator/IdentifierGenerator.php

```diff
@@ -24,5 +24,7 @@
 
 interface IdentifierGenerator
 {
-    public function generate(): string;
+    public const OPTION_RESOURCE_ID = 'resourceId';
+
+    public function generate(array $options = []): string;
```

Comment:
- [ ] @shpran lets please put some annotations to document the options format here and accepted option